internal/apitesting/async.go

```diff
@@ -0,0 +1,159 @@
+package apitesting
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/saucelabs/saucectl/internal/config"
+	"github.com/saucelabs/saucectl/internal/msg"
+	"github.com/saucelabs/saucectl/internal/requesth"
+)
+
+// AsyncResponse describes the json response from the async api endpoints.
+type AsyncResponse struct {
+	ContextIDs []string `json:"contextIds,omitempty"`
+	EventIDs   []string `json:"eventIds,omitempty"`
+	TaskID     string   `json:"taskId,omitempty"`
+	TestIDs    []string `json:"testIds,omitempty"`
+}
+
+// RunAllAsync runs all the tests for the project described by hookId and returns without waiting for their results.
+func (c *Client) RunAllAsync(ctx context.Context, hookId string, buildId string, tunnel config.Tunnel) ([]TestResult, error) {
+	url := c.composeURL(fmt.Sprintf("/api-testing/rest/v4/%s/tests/_run-all", hookId), buildId, "", tunnel)
+
+	req, err := requesth.NewWithContext(ctx, http.MethodPost, url, nil)
+	if err != nil {
+		return []TestResult{}, err
+	}
+
+	req.SetBasicAuth(c.Username, c.AccessKey)
+
+	resp, err := doAsyncRun(c.HTTPClient, req)
+	if err != nil {
+		return []TestResult{}, err
+	}
+
+	apifProject, err := c.GetProject(ctx, hookId)
+
+	if err != nil {
+		apifProject = Project{
+			ID:   "",
+			Name: "",
+		}
+	}
```

Comment:
Do you even need to make the call? 🤔

Reply:
Right. Replaced it with a warning message instead of this empty initialization.